src/CircuitZoo/CircuitZoo.jl

```diff
@@ -132,4 +132,431 @@ function (circuit::Purify2to1)(purifiedL,purifiedR,sacrificedL,sacrificedR)
     success
 end
 
+struct Purify2to1Left <: AbstractCircuit
```

Comment:
could you:

- add a detailed docstring for this circuit, including tests
- rename it to Purify2to1Node (otherwise folks will be wondering where is the `Right` one)